packages/grid/_modules_/grid/models/props/DataGridProProps.ts

```diff
@@ -0,0 +1,139 @@
+  DATA_GRID_PROPS_DEFAULT_VALUES,
+} from './DataGridProps';
+
+/**
+ * The props users can give to the `DataGridProProps` component.
+ */
+export type DataGridProProps = Omit<
+  Partial<DataGridProPropsWithDefaultValue> &
+    DataGridPropsWithComplexDefaultValueBeforeProcessing &
+    DataGridProOtherProps,
+  DataGridProForcedPropsKey
+>;
+
+/**
+ * The props of the `DataGridPro` component after the pre-processing phase.
+ */
+export interface DataGridProProcessedProps
+  extends DataGridProPropsWithDefaultValue,
+    DataGridPropsWithComplexDefaultValueAfterProcessing,
+    DataGridProOtherProps {}
+
+export type DataGridProForcedPropsKey = 'signature';
+
+/**
+ * The `DataGridPro` options with a default value overridable through props
+ * None of the entry of this interface should be optional, they all have default values and `DataGridProps` already applies a `Partial<DataGridSimpleOptions>` for the public interface
+ * The controlled model do not have a default value at the prop processing level, so they must be defined in `DataGridOtherProps`
+ */
+export interface DataGridProPropsWithDefaultValue extends DataGridPropsWithDefaultValues {
+  /**
+   * Set the area in `px` at the bottom of the grid viewport where onRowsScrollEnd is called.
+   * @default 80
+   */
+  scrollEndThreshold: number;
+  /**
+   * If `true`, the rows will be gathered in a tree structure according to the `getTreeDataPath` prop.
+   * @default false
+   */
+  treeData: boolean;
+  /**
+   * If above 0, the row children will be expanded up to this depth.
+   * If equal to -1, all the row children will be expanded.
+   * @default 0
+   */
+  defaultGroupingExpansionDepth: number;
+  /**
+   * Determines if a group should be expanded after its creation.
+   * This prop takes priority over the `defaultGroupingExpansionDepth` prop.
+   * @param {GridRowTreeNodeConfig} node The node of the group to test.
+   * @returns {boolean} A boolean indicating if the group is expanded.
+   */
+  isGroupExpandedByDefault?: (node: GridRowTreeNodeConfig) => boolean;
+  /**
+   * If `true`, the column pinning is disabled.
+   * @default false
+   */
+  disableColumnPinning: boolean;
+  /**
+   * If `true`, the filtering will only be applied to the top level rows.
+   * @default false
+   */
+  disableChildrenFiltering: boolean;
+  /**
+   * If `true`, the sorting will only be applied to the top level rows.
+   * @default false
+   */
+  disableChildrenSorting: boolean;
+}
+
+/**
+ * The default values of `DataGridProSimpleOptions` to inject in the props of DataGridPro.
+ */
+export const DATA_GRID_PRO_PROPS_DEFAULT_VALUES: DataGridProPropsWithDefaultValue = {
+  ...DATA_GRID_PROPS_DEFAULT_VALUES,
+  scrollEndThreshold: 80,
+  treeData: false,
+  defaultGroupingExpansionDepth: 0,
+  disableColumnPinning: false,
+  disableChildrenFiltering: false,
+  disableChildrenSorting: false,
+};
+
+export interface DataGridProOtherProps extends DataGridPropsWithoutDefaultValue {
```

Comment:
It's a long name but more explicit.

```suggestion
export interface DataGridProPropsWithoutDefaultValue extends DataGridPropsWithoutDefaultValue {
```